### ISSUE TITLE
replication module doesn't work!

### ISSUE THREAD

**Issue by meshkin** [eXist-db/messaging-replication#44]
Hi,
I download, biuld, and install this mudule on 2 exist instance (v2.2RC2);
first: localhost:8080/exist (Master)
second : localhost:8082/exist (Slave)
then configured both of them according to guidance in site.
reustart every one and then run this query on master:

``` xquery
let $doc := doc('/db/q.xml')
for $i in (1 to 10)
return
    xmldb:store('/db/repl', concat('mydoc', $i , ".xml"), $doc)
```

but no effect and action applied to slave!
what is mistake?
what related files contains its log?
thanks

**Comment by dizzzz** [comment]
I'll write a 'diagnostic' page later this week; I have my own checklist in case of problems, I'll share it after writing.

In short: logs are in #EXIST-HOME/webapps/WEB-INF/logs/exist.log
- for the slave/receiver, during (early) eXist-db startup there are logs visible, these are there for sure in case of problems
- for the sender/master: I don't know, but if the XAR is loaded correctly, there should be some obvious logging.
- check the Message broker: did you download one and start it? for active mq, if messages are received but not transmitted you'll see messages queued up via the webpage:
  http://miniserver.local:8161/admin/queues.jsp
- known clients are visible via http://miniserver.local:8161/admin/subscribers.jsp 

About how to setup ActiveMq: out of the box it works quite well, for details the webpage should be studied or (better) read the book. it provides a lot of help!

**Comment by meshkin** [comment]
Hi,
Thanks for replies,
I am checking eXist-dbs logs:
after configured eXist-db instances, run activemq with: ./activemq start and it's running...
then started instances. 
both of master and slave has not any logs related to jms, replication, cluster or etc.
however, I run following query on master and stored 10 docs in configured collection:

``` xquery
let $doc := doc('/db/q.xml')
for $i in (1 to 10)
return
    xmldb:store('/db/repl', concat('mydoc', $i , ".xml"), $doc)
```

but had no effect on slave!
apart of master log:

```
2014-08-10 18:27:10,539 [eXistThread-30] DEBUG (TransactionManager.java [execute]:159) - Starting new transaction: 16 
2014-08-10 18:27:10,544 [eXistThread-30] DEBUG (Collection.java [validateXMLResourceInternal]:1631) - Scanning document /db/repl/mydoc1.xml 
2014-08-10 18:27:10,544 [eXistThread-30] DEBUG (ExistSAXParserFactory.java [getSAXParserFactory]:111) - Using default SAXParserFactory 'org.apache.xerces.jaxp.SAXParserFactoryImpl' 
2014-08-10 18:27:10,546 [eXistThread-30] DEBUG (GrammarPool.java [retrieveInitialGrammarSet]:81) - Retrieve initial grammarset (http://www.w3.org/TR/REC-xml). 
2014-08-10 18:27:10,547 [eXistThread-30] DEBUG (GrammarPool.java [retrieveInitialGrammarSet]:85) - Found 0 grammars. 
2014-08-10 18:27:10,554 [eXistThread-30] DEBUG (GrammarPool.java [retrieveInitialGrammarSet]:81) - Retrieve initial grammarset (http://www.w3.org/TR/REC-xml). 
2014-08-10 18:27:10,555 [eXistThread-30] DEBUG (GrammarPool.java [retrieveInitialGrammarSet]:85) - Found 0 grammars. 
2014-08-10 18:27:10,556 [eXistThread-30] DEBUG (Collection.java [storeXMLInternal]:1369) - storing document 6471 ... 
2014-08-10 18:27:10,556 [eXistThread-30] DEBUG (GrammarPool.java [retrieveInitialGrammarSet]:81) - Retrieve initial grammarset (http://www.w3.org/TR/REC-xml). 
2014-08-10 18:27:10,557 [eXistThread-30] DEBUG (GrammarPool.java [retrieveInitialGrammarSet]:85) - Found 0 grammars. 
2014-08-10 18:27:10,576 [eXistThread-30] DEBUG (GrammarPool.java [retrieveInitialGrammarSet]:81) - Retrieve initial grammarset (http://www.w3.org/TR/REC-xml). 
2014-08-10 18:27:10,577 [eXistThread-30] DEBUG (GrammarPool.java [retrieveInitialGrammarSet]:85) - Found 0 grammars. 
2014-08-10 18:27:10,663 [eXistThread-30] DEBUG (Collection.java [storeXMLInternal]:1383) - document stored. 
2014-08-10 18:27:10,663 [eXistThread-30] DEBUG (ReplicationTrigger.java [afterCreateDocument]:146) - /db/repl/mydoc1.xml 
2014-08-10 18:27:10,663 [eXistThread-30] DEBUG (ReplicationTrigger.java [afterUpdateCreateDocument]:104) - /db/repl/mydoc1.xml 
2014-08-10 18:27:10,675 [eXistThread-30] DEBUG (NativeSerializer.java [serializeToReceiver]:112) - serializing document 6471 (/db/repl/mydoc1.xml) to SAX took 9 msec 
2014-08-10 18:27:10,675 [eXistThread-30] DEBUG (PublisherParameters.java [processParameters]:91) - connection.client-id is not set. 
2014-08-10 18:27:10,745 [eXistThread-30] DEBUG (TransactionManager.java [execute]:199) - Committed transaction: 16 
2014-08-10 18:27:10,750 [eXistThread-30] DEBUG (NativeSerializer.java [serializeToReceiver]:112) - serializing document 1485 (/db/q.xml) to SAX took 4 msec
```

on slave, I run exactly following queries:
1)

``` xquery
let $jmsConfiguration := map {
        "java.naming.factory.initial" 
                := "org.apache.activemq.jndi.ActiveMQInitialContextFactory",
        "java.naming.provider.url" := "tcp://localhost:61616",
        "connection-factory" := "ConnectionFactory",
        "destination" := "dynamicTopics/eXistdb-replication-example",
        "subscriber.name" := "SubscriptionId",
        "connection.client-id" := "ClientId"
    }

return
    replication:register($jmsConfiguration)
```

2)

``` xquery
(:
 : Example: send 10 JMS messages with a few message properties
 :)
xquery version "3.0";

import module namespace jms="http://exist-db.org/xquery/messaging" 
              at "java:org.exist.messaging.xquery.MessagingModule";

(: Configuration for setting-up an JMS connection :)
let $jmsConfiguration :=
    map {
        "java.naming.factory.initial" := "org.apache.activemq.jndi.ActiveMQInitialContextFactory",
        "java.naming.provider.url" := "tcp://localhost:61616",
        "destination" := "dynamicQueues/eXistdb-messaging-example",
        "connection-factory" := "ConnectionFactory"
    }

(: JMS message properties :)
let $messageProperties :=
    map {
        "Su" := "Sunday",
        "Mo" := xs:integer(1),
        "Tu" := 2,
        "We" := true(),
        2 := "a",
        "test" := (1,2,3,4,5), (: not transferred :)
        "a" := xs:short(5)      
    }

return
   (: Send 10 messages :)
    for $i in (1 to 10)

    (: The actual message payload :)
    let $content := <data>{$i}</data>

    return
        (: Send message to the JMS broker :)
        jms:send( $content , $messageProperties, $jmsConfiguration ) 
```

3)

``` xquery
(: 
 : Example: register receiver with additional parameters 
 :)
xquery version "3.0";

import module namespace jms="http://exist-db.org/xquery/messaging"  
                        at "java:org.exist.messaging.xquery.MessagingModule";
(: 
 : A (callback) function is required to have 4 parameters. This function
 : logs incoming data to "standard out".
 : 
 : $content The actual data of the messages
 : $params  Additional parameters set when registerering the callback function
 : $messageProperties JMS message properties, contains data of sender and 
 :                    the eXist-db database
 : $jmsConfig Details about the JMS configuration (for debugging) 
 :)
declare function local:handleMessage($content as item(), $params as item()*, 
                            $messageProperties as map(), $jmsConfig as map() )
{
    ( 
        util:log-system-out( util:get-sequence-type($content) 
                             || "->" || data($content) ), 

        util:log-system-out( for $param in $params
                             return util:get-sequence-type($param) 
                             || "->" || $param || " ;  " ), 

        util:log-system-out( 
            for $key in map:keys($messageProperties)
            let $value := map:get($messageProperties, $key) 
            return util:get-sequence-type($value) || "-->" || $key || "=" ||  $value || " ;  " ), 

        util:log-system-out( for $key in map:keys($jmsConfig)
            let $value := map:get($jmsConfig, $key) 
            return util:get-sequence-type($value) || "-->" || $key || "=" ||  $value || " ;  " )
    )
};
(: Configuration for setting-up an JMS connection :)
let $jmsConfiguration :=
    map {
        "java.naming.factory.initial" := "org.apache.activemq.jndi.ActiveMQInitialContextFactory",
        "java.naming.provider.url" := "tcp://localhost:61616",
        "destination" := "dynamicQueues/eXistdb-messaging-example",
        "connection-factory" := "ConnectionFactory"
    }

(: Define HoF callback function, with 4 parameters :)
let $callback := local:handleMessage#4 

(: Additional (optional) data to parameterize the callback function :)
let $additionalParameters := (1, "2" , xs:float(3.0))

return
    (: Register the function to the JMS broker :)
    jms:register( $callback , $additionalParameters, $jmsConfiguration)
```

and logged:

```
2014-08-10 18:21:44,876 [eXistThread-30] INFO  (ReceiversManager.java [register]:66) - Registering receiver 1 
2014-08-10 18:21:44,876 [eXistThread-30] INFO  (Receiver.java [initialize]:167) - Initializing JMS connection 
2014-08-10 18:21:45,217 [eXistThread-30] INFO  (Receiver.java [initialize]:231) - JMS connection is initialized: connection.client-id=ClientId 
2014-08-10 18:21:45,218 [eXistThread-30] INFO  (Receiver.java [start]:149) - JMS connection is started. ClientId=ClientId 
2014-08-10 18:21:45,371 [eXistThread-30] INFO  (TemporaryFileManager.java [<init>]:76) - Temporary folder is: /home/test/eXist-db/tools/jetty/tmp/_mmtfm_e4d44e21-fdb7-45b3-ad50-9c7cfbce0fed 
2014-08-10 18:21:59,324 [eXistThread-32] INFO  (MessagingConfiguration.java [loadConfiguration]:94) - Cannot convert a sequence of values for key 'test' into JMS message properties 
2014-08-10 18:21:59,533 [eXistThread-32] INFO  (MessagingConfiguration.java [loadConfiguration]:94) - Cannot convert a sequence of values for key 'test' into JMS message properties 
2014-08-10 18:21:59,581 [eXistThread-32] INFO  (MessagingConfiguration.java [loadConfiguration]:94) - Cannot convert a sequence of values for key 'test' into JMS message properties 
2014-08-10 18:21:59,632 [eXistThread-32] INFO  (MessagingConfiguration.java [loadConfiguration]:94) - Cannot convert a sequence of values for key 'test' into JMS message properties 
2014-08-10 18:21:59,700 [eXistThread-32] INFO  (MessagingConfiguration.java [loadConfiguration]:94) - Cannot convert a sequence of values for key 'test' into JMS message properties 
2014-08-10 18:21:59,759 [eXistThread-32] INFO  (MessagingConfiguration.java [loadConfiguration]:94) - Cannot convert a sequence of values for key 'test' into JMS message properties 
2014-08-10 18:21:59,816 [eXistThread-32] INFO  (MessagingConfiguration.java [loadConfiguration]:94) - Cannot convert a sequence of values for key 'test' into JMS message properties 
2014-08-10 18:21:59,867 [eXistThread-32] INFO  (MessagingConfiguration.java [loadConfiguration]:94) - Cannot convert a sequence of values for key 'test' into JMS message properties 
2014-08-10 18:21:59,920 [eXistThread-32] INFO  (MessagingConfiguration.java [loadConfiguration]:94) - Cannot convert a sequence of values for key 'test' into JMS message properties 
2014-08-10 18:21:59,987 [eXistThread-32] INFO  (MessagingConfiguration.java [loadConfiguration]:94) - Cannot convert a sequence of values for key 'test' into JMS message properties 
2014-08-10 18:22:41,797 [DefaultQuartzScheduler_Worker-1] INFO  (NativeBroker.java [sync]:3669) - Memory: 227,840K total; 932,352K max; 133,446K free 
2014-08-10 18:22:41,797 [DefaultQuartzScheduler_Worker-1] INFO  (BTree.java [printStatistics]:2387) - dom.dbx INDEX Buffers occupation : 12% (8 out of 64) Cache efficiency : 98% 
2014-08-10 18:22:41,797 [DefaultQuartzScheduler_Worker-1] INFO  (DOMFile.java [printStatistics]:1369) - dom.dbx DATA Buffers occupation : 5% (14 out of 256) Cache efficiency : 99% 
2014-08-10 18:22:41,797 [DefaultQuartzScheduler_Worker-1] INFO  (BTree.java [printStatistics]:2387) - collections.dbx INDEX Buffers occupation : 17% (11 out of 64) Cache efficiency : 91% 
2014-08-10 18:22:41,797 [DefaultQuartzScheduler_Worker-1] INFO  (BFile.java [printStatistics]:434) - collections.dbx DATA Buffers occupation : 19% (12 out of 64) Cache efficiency : 87% 
2014-08-10 18:22:41,798 [DefaultQuartzScheduler_Worker-1] INFO  (BTree.java [printStatistics]:2387) - values.dbx INDEX Buffers occupation : 0% (0 out of 64) Cache efficiency : N/A 
2014-08-10 18:22:41,798 [DefaultQuartzScheduler_Worker-1] INFO  (BFile.java [printStatistics]:434) - values.dbx DATA Buffers occupation : 0% (0 out of 64) Cache efficiency : N/A 
2014-08-10 18:23:24,190 [eXistThread-29] INFO  (ReceiversManager.java [register]:66) - Registering receiver 2 
2014-08-10 18:23:24,190 [eXistThread-29] INFO  (Receiver.java [initialize]:167) - Initializing JMS connection 
2014-08-10 18:23:24,206 [eXistThread-29] INFO  (Receiver.java [initialize]:231) - JMS connection is initialized: connection.client-id=ID:localhost.localdomain-42755-1407678705040-22:1 
2014-08-10 18:23:24,207 [eXistThread-29] INFO  (Receiver.java [start]:149) - JMS connection is started. ClientId=ID:localhost.localdomain-42755-1407678705040-22:1 
2014-08-10 18:23:24,225 [ActiveMQ Session Task-1] INFO  (MessagingJmsListener.java [onMessage]:133) - {2} Received message: messageId=ID:localhost.localdomain-42755-1407678705040-3:1:1:1:1 
2014-08-10 18:23:24,279 [ActiveMQ Session Task-1] INFO  (MessagingJmsListener.java [onMessage]:133) - {2} Received message: messageId=ID:localhost.localdomain-42755-1407678705040-5:1:1:1:1 
2014-08-10 18:23:24,287 [ActiveMQ Session Task-1] INFO  (MessagingJmsListener.java [onMessage]:133) - {2} Received message: messageId=ID:localhost.localdomain-42755-1407678705040-7:1:1:1:1 
2014-08-10 18:23:24,295 [ActiveMQ Session Task-1] INFO  (MessagingJmsListener.java [onMessage]:133) - {2} Received message: messageId=ID:localhost.localdomain-42755-1407678705040-9:1:1:1:1 
2014-08-10 18:23:24,304 [ActiveMQ Session Task-1] INFO  (MessagingJmsListener.java [onMessage]:133) - {2} Received message: messageId=ID:localhost.localdomain-42755-1407678705040-11:1:1:1:1 
2014-08-10 18:23:24,334 [ActiveMQ Session Task-1] INFO  (MessagingJmsListener.java [onMessage]:133) - {2} Received message: messageId=ID:localhost.localdomain-42755-1407678705040-13:1:1:1:1 
2014-08-10 18:23:24,360 [ActiveMQ Session Task-1] INFO  (MessagingJmsListener.java [onMessage]:133) - {2} Received message: messageId=ID:localhost.localdomain-42755-1407678705040-15:1:1:1:1 
2014-08-10 18:23:24,384 [ActiveMQ Session Task-1] INFO  (MessagingJmsListener.java [onMessage]:133) - {2} Received message: messageId=ID:localhost.localdomain-42755-1407678705040-17:1:1:1:1 
2014-08-10 18:23:24,392 [ActiveMQ Session Task-1] INFO  (MessagingJmsListener.java [onMessage]:133) - {2} Received message: messageId=ID:localhost.localdomain-42755-1407678705040-19:1:1:1:1 
2014-08-10 18:23:24,400 [ActiveMQ Session Task-1] INFO  (MessagingJmsListener.java [onMessage]:133) - {2} Received message: messageId=ID:localhost.localdomain-42755-1407678705040-21:1:1:1:1 
```

note: I installed instances on one system! but they running on diffrent ports.
excuse me if comment is too long.

**Comment by dizzzz** [comment]
Please make sure you don't mix up messaging (=sending messages) and replication (=syncing documents);

regarding replication:
- for this you need to define a collection.xconf file on the 'master' ; can you share this file? 
- I see "ReplicationTrigger.java" which means documents are actually send out
- I asked you to check the web-interface of active MQ: are there any messages queued in the topics?
- I need to recheck if "connection.client-id" is required; probably not since I put it on DEBUG level and not ERROR

regarding messaging:
- you do send/receive of messages on the same exist-db ; with the "MessagingJmsListener.java" it shows that the function is working

**Comment by dizzzz** [comment]
please note there is a error in the wiki example:

on https://github.com/eXist-db/messaging-replication/wiki/Document%20replication there are two different dynamicTopics defined:

`dynamicTopics/eXistdb-replication-example` (master/slave)
and
`dynamicTopics/eXistdb-replication-demo` (xquery example)

that might be the reason for it not to be working. Apologies for the documentation bug :-( I have updated the text. It is `dynamicTopics/eXistdb-replication-example` now.

I am not sure if it solves your issue though. please check the activemq management console for connected clients / messages in the Topics section.

**Comment by rhhaley** [comment]
Did you extract the xar file into exist/lib/user

**Comment by dizzzz** [comment]
That should not be required anymore, when the replication-receiver is started via an xquery script; this script can be automatically started during exist-db booting... 

https://github.com/eXist-db/exist/blob/develop/conf.xml.tmpl#L180

this needs to be enabled manually; 2.2RC1 and newer

**Comment by meshkin** [comment]
Hi,
I Created the collection.xconf file in /db/system/config/db/repl collection with following contents:

``` xml
<collection xmlns="http://exist-db.org/collection-config/1.0">
    <triggers>
        <trigger class="org.exist.replication.jms.publish.ReplicationTrigger">
            <!-- 
                Class name of the initial context provider, default value 
                for ActiveMQ
                see javax.naming.Context#INITIAL_CONTEXT_FACTORY
            -->
            <parameter name="java.naming.factory.initial" value="org.apache.activemq.jndi.ActiveMQInitialContextFactory"/>
            <!-- 
                URL of the message broker, default value for ActiveMQ 
                see javax.naming.Context#PROVIDER_URL
            -->
            <parameter name="java.naming.provider.url" value="tcp://localhost:61616"/>
            <!-- 
                Lookup connection factory
                see javax.naming.InitialContext#lookup(String) 
            -->
            <parameter name="connection-factory" value="ConnectionFactory"/>
            <!--
                Lookup destination (topic)
                see javax.naming.InitialContext#lookup(String) 
            -->
            <parameter name="destination" value="dynamicTopics/eXistdb-replication-example"/>
        </trigger>
    </triggers>
</collection>
```

Sorry, I forgot report activemq logs!
today, I run both of instances after run activemq server; then delete 10 docs from repl collection on master and create them again. the activemq web-pages are in bellow:
![queues](https://cloud.githubusercontent.com/assets/910083/3874461/0e194ebe-2142-11e4-9a55-49ba80e3c9ad.png)
![topics](https://cloud.githubusercontent.com/assets/910083/3874471/1b07ea5e-2142-11e4-93b8-85a9a102117c.png)
and other its web-pages not showing any thing!

as well as, after that, I run the register query, then send query. result:

``` xml
<jms>

<jms.message-id>ID:localhost.localdomain-53767-1407752937692-3:1:1:1:1</jms.message-id>
<producer.time-to-live>0</producer.time-to-live>
<producer.priority>0</producer.priority>
<java.naming.factory.initial>org.apache.activemq.jndi.ActiveMQInitialContextFactory</java.naming.factory.initial>
<java.naming.provider.url>tcp://localhost:61616</java.naming.provider.url>
<connection-factory>ConnectionFactory</connection-factory>
<destination>dynamicQueues/eXistdb-messaging-example</destination>
</jms>

<jms>
<jms.message-id>ID:localhost.localdomain-53767-1407752937692-5:1:1:1:1</jms.message-id>
<producer.time-to-live>0</producer.time-to-live>
<producer.priority>0</producer.priority>
<java.naming.factory.initial>org.apache.activemq.jndi.ActiveMQInitialContextFactory</java.naming.factory.initial>
<java.naming.provider.url>tcp://localhost:61616</java.naming.provider.url>
<connection-factory>ConnectionFactory</connection-factory>
<destination>dynamicQueues/eXistdb-messaging-example</destination>
</jms>

<jms>
<jms.message-id>ID:localhost.localdomain-53767-1407752937692-7:1:1:1:1</jms.message-id>
<producer.time-to-live>0</producer.time-to-live>
<producer.priority>0</producer.priority>
<java.naming.factory.initial>org.apache.activemq.jndi.ActiveMQInitialContextFactory</java.naming.factory.initial>
<java.naming.provider.url>tcp://localhost:61616</java.naming.provider.url>
<connection-factory>ConnectionFactory</connection-factory>
<destination>dynamicQueues/eXistdb-messaging-example</destination>
</jms>

<jms>
<jms.message-id>ID:localhost.localdomain-53767-1407752937692-9:1:1:1:1</jms.message-id>
<producer.time-to-live>0</producer.time-to-live>
<producer.priority>0</producer.priority>
<java.naming.factory.initial>org.apache.activemq.jndi.ActiveMQInitialContextFactory</java.naming.factory.initial>
<java.naming.provider.url>tcp://localhost:61616</java.naming.provider.url>
<connection-factory>ConnectionFactory</connection-factory>
<destination>dynamicQueues/eXistdb-messaging-example</destination>
</jms>

<jms>
<jms.message-id>ID:localhost.localdomain-53767-1407752937692-11:1:1:1:1</jms.message-id>
<producer.time-to-live>0</producer.time-to-live>
<producer.priority>0</producer.priority>
<java.naming.factory.initial>org.apache.activemq.jndi.ActiveMQInitialContextFactory</java.naming.factory.initial>
<java.naming.provider.url>tcp://localhost:61616</java.naming.provider.url>
<connection-factory>ConnectionFactory</connection-factory>
<destination>dynamicQueues/eXistdb-messaging-example</destination>
</jms>

<jms>
<jms.message-id>ID:localhost.localdomain-53767-1407752937692-13:1:1:1:1</jms.message-id>
<producer.time-to-live>0</producer.time-to-live>
<producer.priority>0</producer.priority>
<java.naming.factory.initial>org.apache.activemq.jndi.ActiveMQInitialContextFactory</java.naming.factory.initial>
<java.naming.provider.url>tcp://localhost:61616</java.naming.provider.url>
<connection-factory>ConnectionFactory</connection-factory>
<destination>dynamicQueues/eXistdb-messaging-example</destination>
</jms>

<jms>
<jms.message-id>ID:localhost.localdomain-53767-1407752937692-15:1:1:1:1</jms.message-id>
<producer.time-to-live>0</producer.time-to-live>
<producer.priority>0</producer.priority>
<java.naming.factory.initial>org.apache.activemq.jndi.ActiveMQInitialContextFactory</java.naming.factory.initial>
<java.naming.provider.url>tcp://localhost:61616</java.naming.provider.url>
<connection-factory>ConnectionFactory</connection-factory>
<destination>dynamicQueues/eXistdb-messaging-example</destination>
</jms>

<jms>
<jms.message-id>ID:localhost.localdomain-53767-1407752937692-17:1:1:1:1</jms.message-id>
<producer.time-to-live>0</producer.time-to-live>
<producer.priority>0</producer.priority>
<java.naming.factory.initial>org.apache.activemq.jndi.ActiveMQInitialContextFactory</java.naming.factory.initial>
<java.naming.provider.url>tcp://localhost:61616</java.naming.provider.url>
<connection-factory>ConnectionFactory</connection-factory>
<destination>dynamicQueues/eXistdb-messaging-example</destination>
</jms>

<jms>
<jms.message-id>ID:localhost.localdomain-53767-1407752937692-19:1:1:1:1</jms.message-id>
<producer.time-to-live>0</producer.time-to-live>
<producer.priority>0</producer.priority>
<java.naming.factory.initial>org.apache.activemq.jndi.ActiveMQInitialContextFactory</java.naming.factory.initial>
<java.naming.provider.url>tcp://localhost:61616</java.naming.provider.url>
<connection-factory>ConnectionFactory</connection-factory>
<destination>dynamicQueues/eXistdb-messaging-example</destination>
</jms>

<jms>
<jms.message-id>ID:localhost.localdomain-53767-1407752937692-21:1:1:1:1</jms.message-id>
<producer.time-to-live>0</producer.time-to-live>
<producer.priority>0</producer.priority>
<java.naming.factory.initial>org.apache.activemq.jndi.ActiveMQInitialContextFactory</java.naming.factory.initial>
<java.naming.provider.url>tcp://localhost:61616</java.naming.provider.url>
<connection-factory>ConnectionFactory</connection-factory>
<destination>dynamicQueues/eXistdb-messaging-example</destination>
</jms>
```

and then receive and report query. its result:

``` xml
<Receiver id="1">
<state>STARTED</state>
<java.naming.factory.initial>org.apache.activemq.jndi.ActiveMQInitialContextFactory</java.naming.factory.initial>
<java.naming.provider.url>tcp://localhost:61616</java.naming.provider.url>
<connection-factory>ConnectionFactory</connection-factory>
<destination>dynamicTopics/eXistdb-replication-example</destination>
<connection.client-id>ClientId</connection.client-id>
<usage>replication</usage>
<errorMessages/>
<statistics>
<nrProcessedMessages>0</nrProcessedMessages>
<cumulativeProcessingTime>PT0.000S</cumulativeProcessingTime>
<nrFailedMessages>0</nrFailedMessages>
</statistics>
</Receiver>

<Receiver id="2">
<state>STARTED</state>
<java.naming.factory.initial>org.apache.activemq.jndi.ActiveMQInitialContextFactory</java.naming.factory.initial>
<java.naming.provider.url>tcp://localhost:61616</java.naming.provider.url>
<connection-factory>ConnectionFactory</connection-factory>
<destination>dynamicQueues/eXistdb-messaging-example</destination>
<connection.client-id>ID:localhost.localdomain-53767-1407752937692-22:1</connection.client-id>
<usage>messaging</usage>
<errorMessages/>
<statistics>
<nrProcessedMessages>10</nrProcessedMessages>
<cumulativeProcessingTime>PT0.191S</cumulativeProcessingTime>
<nrFailedMessages>0</nrFailedMessages>
</statistics>
</Receiver>
```

the activemq updated with as follows:
![7777](https://cloud.githubusercontent.com/assets/910083/3874579/74a71d90-2143-11e4-941b-2e5df8c8083c.png)
![88888](https://cloud.githubusercontent.com/assets/910083/3874576/749d210a-2143-11e4-8ed0-eb4dc7b05682.png)
![99999](https://cloud.githubusercontent.com/assets/910083/3874578/74a5d142-2143-11e4-9a64-dde4d7d9c06b.png)
![101010](https://cloud.githubusercontent.com/assets/910083/3874577/74a4921e-2143-11e4-9648-6ae382144c08.png)

In the exist.log is mentioned that client-id not set. I guess the problem is caused by it (in 2013 slides, in collection.xconf there is the client-id). isn't true?

**Comment by dizzzz** [comment]
about client id: I need to verify this, it is some time ago, but I remember I did add some comments about this in the wiki documentation. it is always possible to set it.....but when no errors are shown not required.

you messages are a bit confusing to me: are we chasing replication or messaging? we MUST stay on one (1) topic otherwise support is too hard to handle

did you install the gui-xar file btw? it provides some statistics as well; it will download the other xar as well though....

**Comment by dizzzz** [comment]
the reporting suggests:
- no incoming replication event : not configured correct? client-id?
- messages are coming in (10x)

for the rest.. things look ok....

**Comment by meshkin** [comment]
HI,
After I ran the ant install command, the following files were made in dist directory:
messaging-replication-develop/dist/messaging-replication-core-0.7.5.xar
messaging-replication-develop/dist/messaging-replication-ui-0.7.5.xar
then, I went to "Admin Web Application -> Packages Repository" and upload them. then they were added to installed packages.
too, I copied the jar files wihitn them to EXIST_HOME/lib/user and then build and configure replication (according documantion) for every exist instance and build eXist instnces, again.
but not work yet!!

**Comment by dizzzz** [comment]
A few points, step by step
1. The goal is still replication right?
2. Version 0.7.5 is not released yet, if possible use the downloadable 0.7.4 XARs
3. copying the JAR files to `EXIST_HOME/lib/user` is ply required if you want to start the replication listener during database startup; for now, don't copy the JAR files, start the listener via XQuery manually 
4. are there any more log messages?
5. During tests, please delete the topics/queues from activemq ; that will make it more clear what is happening. ideally the nr of queued and dequeued messages in the 'topics' should be equally; meaning: all files are delivered.
6. Do you see the replication items appear in the activemq webui ?
- if so, then the issue is the slave not the master (probably the case based on the images)
- if not that the issue is at the master not the slave

**Comment by dizzzz** [comment]
no response, closing ticket. (can be reopenend)
